src/TAProcessor.cpp

```diff
@@ -102,24 +105,50 @@ TAProcessor::conf(const appmodel::DataHandlerModule* conf)
 //   //ci.add(info);
 // }
 
+void
+TAProcessor::generate_opmon_data()
+{
+  opmon::TAProcessorInfo info;
+
+  info.set_ta_received_count( m_ta_received_count.load() );
+  info.set_tc_made_count( m_tc_made_count.load() );
+  info.set_tc_sent_count( m_tc_sent_count.load() );
+  info.set_tc_failed_sent_count( m_tc_failed_sent_count.load() );
+
+  this->publish(std::move(info));
+}
 
 /**
  * Pipeline Stage 2.: Do software TPG
  * */
 void
 TAProcessor::find_tc(const TAWrapper* ta,  std::shared_ptr<triggeralgs::TriggerCandidateMaker> tca)
 {
+  m_ta_received_count++;
   std::vector<triggeralgs::TriggerCandidate> tcs;
   tca->operator()(ta->activity, tcs);
   for (auto tc : tcs) {
+    m_tc_made_count++;
     if(!m_tc_sink->try_send(std::move(tc), iomanager::Sender::s_no_block)) {
         ers::warning(TCDropped(ERS_HERE, tc.time_start, m_sourceid.id));
-        m_tcs_dropped++;
+        m_tc_failed_sent_count++;
     }
-    m_new_tcs++;
+    m_tc_sent_count++;
```

Comment:
```suggestion
    else {
        m_tc_sent_count++;
    }
```

I think this needs to be inside of an else statement, otherwise the TCs that failed to send will always simultaneously count as sent.

Previously this was a counter for number of TCs made, so that was OK. Now it looks like that counter moved up to 131, and this one only should correspond to the TCs that were successfully sent?

Reply:
Alternatively, you can add 
`continue;` after `m_tc_failed_sent_count++;` line, so the `m_tc_sent_count++` doesn't get triggered if tc was not sent.